src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,181 @@
+            }
+            Todo newTask = new Todo(userInput);
+            tasks.add(newTask);
+            printTaskAddedConfirmation(newTask);
+            updateFile();
+        } catch (InvalidInputException e){
+            System.out.println(e.toString().substring(23));
+        }
+    }
+
+    public static void addDeadline(String userInput) {
+        try {
+            if (userInput.length()<9){
+                throw new InvalidInputException("OOPS!!! Description of deadline cannot be empty :(");
+            }
+            if (!userInput.contains("/by ")){
+                throw new InvalidInputException("OOPS!! Please input date of deadline");
+            }
+            int startIndexOfTask = userInput.indexOf(' ')+1;
+            int endIndexOfTask = userInput.indexOf('/') - 1;
+
+            int startIndexOfDeadline = userInput.indexOf('/') + 4;
+            String taskName = userInput.substring(startIndexOfTask, endIndexOfTask);
+            String deadline = userInput.substring(startIndexOfDeadline);
+            Deadline newTask = new Deadline(taskName, deadline);
+            tasks.add(newTask);
+            printTaskAddedConfirmation(newTask);
+            updateFile();
+        } catch (InvalidInputException e){
+            System.out.println(e.toString().substring(23));
+        }
+    }
+
+    public static void addEvent(String userInput) {
+        try {
+            if (userInput.length()<6) {
+                throw new InvalidInputException("OOPS!!! Description of event cannot be empty :(");
+            }
+            if (!userInput.contains("/at ")){
+                throw new InvalidInputException("OOPS!! Please input time of event");
+            }
+            int startIndexOfTask = userInput.indexOf(' ')+1;
+            int endIndexOfTask = userInput.indexOf('/') - 1;
+            int startIndexOfDate = userInput.indexOf('/') + 4;
+            String taskName = userInput.substring(startIndexOfTask, endIndexOfTask);
+            String deadline = userInput.substring(startIndexOfDate);
+            Event newTask = new Event(taskName, deadline);
+            tasks.add(newTask);
+            printTaskAddedConfirmation(newTask);
+            updateFile();
+        } catch (InvalidInputException e){
+            System.out.println(e.toString().substring(23));
+        }
+
+    }
+
+    public static void printTasks() {
+        if (tasks.size() == 0) {
+            System.out.println("\tNo tasks added");
+            return;
+        }
+        int count = 1;
+        for (Todo task : tasks) {
+            System.out.println("\tHere are the tasks in your list");
+            System.out.println("\t"+count+ "." + task);
+            count++;
+        }
+    }
+
+    public static void markDone(String userInput) {
+        int indexOfTaskNumber = userInput.indexOf(' ') + 1;
+        int taskNumber = Integer.parseInt(userInput.substring(indexOfTaskNumber));
+        if (taskNumber > tasks.size() || taskNumber <= 0) {
```

Comment:
The same code is repeated in several other lines in this file. You might want to consider abstracting out this logic into a validation helper function